Restrict FPGA tests to Arduino form factor

fkjagodzinski opened the pull request:
Currently only the Arduino form factor is supported (support for other form factors will be added in the future).

This patch will prevent the compilation of greentea tests for targets with non-arduino form factors (or no form factos at all). In consequence, this patch will fix compilation errors for `EFM32GG11_STK3701` and `TB_SENSE_12` when `FPGA_CI_TEST_SHIELD` component is added for all targets (https://github.com/ARMmbed/mbed-os/pull/10965#issuecomment-508734380).

### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->
@c1728p9 @mprse @maciejbocianski @jamesbeyond 

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

fkjagodzinski committed:
FPGA test shield: Restrict tests to Arduino FF only

Currently only the Arduino form factor is supported. Support for other
form factors will be added in the future.

maciejbocianski reviewed (approved):
LGTM

c1728p9 commented on TESTS/mbed_hal_fpga_ci_test_shield/analogin/main.cpp:
```diff
@@ -19,6 +19,8 @@
 #error [NOT_SUPPORTED] Analog in not supported for this target
 #elif !COMPONENT_FPGA_CI_TEST_SHIELD
 #error [NOT_SUPPORTED] FPGA CI Test Shield is needed to run this test
+#elif !TARGET_FF_ARDUINO
```
Rather than `!TARGET_FF_ARDUINO` this should be `!defined(MBED_CONF_TARGET_DEFAULT_FORM_FACTOR)`. This macro is set whenever there is a [valid form factor](https://github.com/ARMmbed/mbed-os/blob/master/hal/pinmap.h#L166-L196).

c1728p9 committed:
FPGA test shield: Allow any defined form factor

Build in the FPGA tests if either MBED_CONF_TARGET_DEFAULT_FORM_FACTOR
or TARGET_FF_ARDUINO is defined.
